### DISCUSSION TITLE
DataFrame.to_html() inside <html> code?

### DISCUSSION THREAD

**Discussion by dlink1999** [deta/cloud-docs#235] [Help] [answered]
Dear All:
Does anyone know how to use escape code, or other method to put the <table>...</table> generated by DataFrame.to_html() inside the html code below? Thanks. 

    return """
    <html>
        <head>
            <title>Some HTML in here</title>
        </head>
        <body>
            <h1>Look ma! HTML!</h1>
            {*** DataFrame.to_html() here***}
        </body>
    </html>
    """

**Comment by jnsougata** [comment]
```html
<div class="right_container">
    {% for data in all_info %}
        <div class="right_content" onclick="window.open('{{data['url']}}')">
            {{data['name']}} - {{receivers[data['id']]}}
        </div>
    {% endfor %}

</div>
```

This is an example of running python code inside HTML. 
For reference see: 
1.`Using Jinja2Templates` https://fastapi.tiangolo.com/advanced/templates/#using-jinja2templates 
2.`Writing templates` https://fastapi.tiangolo.com/advanced/templates/#writing-templates

**Reply by dlink1999** [reply]
Thanks. I need to study Jinja to understand the solution.

**Comment by abdelhai** [comment, marked as answer]
1. generate the html already inside your python code & use jinja to render it
2. render the html markup using jinja's `safe`: https://jinja.palletsprojects.com/en/3.1.x/templates/#working-with-manual-escaping

**Reply by dlink1999** [reply]
Thanks. I need to study Jinja to understand the solution
